portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java

```diff
@@ -103,19 +105,19 @@ private void pollForUnassignedJob() throws IOException {
      * Updates a unassigned {@link PortabilityJob} in storage with the provided {@code jobId} in
      * CREDS_ENCRYPTION_KEY_GENERATED state with {@code publicKey} and {@code privateKey}.
      */
-    private void keyGenerated(UUID jobId, PublicKey publicKey, PrivateKey privateKey)
+    private void keyGenerated(UUID jobId, PublicKey publicKey)
             throws IOException {
         // Lookup the job so we can append to its existing properties.
         PortabilityJob existingJob = store.findJob(jobId);
         // Verify no worker key
         // Populate job with keys to persist
-        String encodedPublicKey = PublicPrivateKeyPairGenerator.encodeKey(publicKey);
-        String encodedPrivateKey = PublicPrivateKeyPairGenerator.encodeKey(privateKey);
+        Preconditions.checkArgument(existingJob.jobAuthorization().encryptedPublicKey() == null);
```

Comment:
can you add a message, worker key should not be present in the PortabilityJob yet, as we're about to set it to the key we generated just now

Reply:
Done